akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -129,12 +129,11 @@ object ActorMaterializer {
    * INTERNAL API
    */
   private[akka] def downcast(materializer: Materializer): ActorMaterializer =
-    materializer match {
+    materializer match { //FIXME this method is going to cause trouble for other Materializer implementations
```

Comment:
This has the signature of a responsibility problem, if something on the outside needs to know whether something is an ActorMaterializer or not and -fail- if it isn't, then I'd argue that it's not being honest about its dependencies.
Materialization should be contained in the Materializer, perhaps we need to take a step back and see if a restructuing would eliminate the need for downcasting.